hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HandleHederaOperations.java

```diff
@@ -186,8 +187,13 @@ public long lazyCreationCostInGas(@NonNull final Address recipient) {
         final var synthCreation = TransactionBody.newBuilder()
                 .cryptoCreateAccount(CREATE_TXN_BODY_BUILDER.alias(tuweniToPbjBytes(recipient)))
                 .build();
-        final var createFee = gasCalculator.canonicalPriceInTinybars(synthCreation, payerId);
-        return (createFee) / gasCalculator.topLevelGasPrice();
+        final var createFee = gasCalculator.feeCalculatorPriceInTinyBars(synthCreation, payerId);
+
+        // isGasPrecisionLossFixEnabled is a temporary feature flag that will be removed in the future.
+        if (!contractsConfig.isGasPrecisionLossFixEnabled()) {
+            return (createFee) / gasCalculator.topLevelGasPrice();
+        }
+        return (createFee) * FEE_SCHEDULE_UNITS_PER_TINYCENT / gasCalculator.topLevelGasPriceInTinyBars();
```

Comment:
Do we not want to do some sort of compare here in case the canonical price is more?

Reply:
This is how it was before. I guess we could reconsider how the pricing is structured, however I have a gut feeling this could end up costing a looot of gas.

Reply:
This is a great point. To address the confusion: previously, the name `canonicalPriceInTinybars` was very misleading as it was not the canonical price. That's why the method now says `feeCalculatorPriceInTinyBars`.

However, I think this should definitely be checked, as from what I debugged, just now, it seems that `feeCalculatorPriceInTinyBars` returns 0, so this may be another flaw in the previous system, as it seems to have returned 0 before as well.

Careful consideration should be given, even though it seems like a flaw, as this may increase the gas by hundreds of thousands if we decide to go with the canonical price.

Reply:
Added this issue to address at a future time.
https://github.com/hashgraph/hedera-services/issues/15337